process/block/shardblock.go

```diff
@@ -1363,17 +1483,22 @@ func (sp *shardProcessor) createMiniBlocks(
 		log.Info(err.Error())
 	}
 
+	processedMetaHdrs, errNotCritical := sp.getProcessedMetaBlocks(destMeMiniBlocks, usedMetaHdrsHashes)
+	if errNotCritical != nil {
+		log.Debug(errNotCritical.Error())
+	}
+
+	err = sp.setMetaConsensusData(processedMetaHdrs)
```

Comment:
a separate component could get the usedMetaHdrHashes and the dstMeMiniBlocks and set all the needed data to the special address handler. Should me moved out from blockprocessor.

Reply:
the logic for selecting the right addresses etc was moved to the address handler, these simply call the interface, so there is no need of creating a separate component.